### PR TITLE
Make auth required easy to mock

### PR DESCRIPTION
**Pull request by bmbouter**
This PR will allow auth_required statement able to easily have assertions made about them.

Docs on how to use this can be found here:  https://fedorahosted.org/pulp/wiki/ReplacewebpywithDjango#UnitTesting

### CODE REVIEW

**Comment by rbarlow** [comment]
Hey that's nice!

File: server/pulp/server/auth/authorization.py [hunk 1]
```diff
@@ -1,19 +1,5 @@
...
-# 2 of the License (GPLv2) or (at your option) any later version.
-# There is NO WARRANTY for this software, express or implied,
-# including the implied warranties of MERCHANTABILITY,
-# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
-# have received a copy of GPLv2 along with this software; if not, see
-# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
-
-"""
-Utility functions to manage permissions and roles in pulp.
-"""
```
Comment (rbarlow) [inline server/pulp/server/auth/authorization.py]: PEP-257 requires modules to have docblocks too, and once I finish PEP-8, PEP-257 is next. Plz to add back ☺

**Comment by bmbouter** [comment]
I think I need to better handle the case when @auth_required(). Fixing it now.

File: server/pulp/server/auth/authorization.py (hunk 1 above)
Comment (bmbouter) [inline server/pulp/server/auth/authorization.py]: I did not know that. I will bring it back with the next commit.

**Comment by bmbouter** [comment]
I documented how to use this in the porting to Django guide.

https://fedorahosted.org/pulp/wiki/ReplacewebpywithDjango#UnitTesting

**Comment by bmbouter** [comment]
I will squash this before I merge it.

**Comment by asmacdo** [comment]
Very clever @bmbouter! Also simple to use and will require minimal changes. 

I have only done minimal hand testing, but we will be putting it through the ringer as we merge the PR backlog, so I think we should go ahead an merge it.  LGTM. 

I think we should also add unit tests for `_verify_auth`, but I am ok with adding this later so we can begin the merges.

**Comment by bmbouter** [comment]
Squashed and Merging.

**Pull request merged**